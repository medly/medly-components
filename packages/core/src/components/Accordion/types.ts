export type StaticProps = {
    Header: React.FC;
    Content: React.FC;
    Context: React.Context<AccordionContextType>;
};

export type AccordionContextType = [
    boolean /** Current active state */,
    React.Dispatch<React.SetStateAction<boolean>> /** Function to be called to toggle the active state */
];
