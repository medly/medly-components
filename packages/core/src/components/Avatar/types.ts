import { AvatarSizes } from '@medly-components/theme';
import { HTMLProps, Omit } from '@medly-components/utils';

export interface AvatarProps extends Omit<HTMLProps<HTMLDivElement>, 'size'> {
    /** Size of the avatar */
    size?: AvatarSizes;
    /** Text color */
    textColor?: string;
    /** Background color */
    bgColor?: string;
    /** Hover Text color */
    hoverTextColor?: string;
    /** Hover Background color */
    hoverBgColor?: string;
    /** Hover text shadow color*/
    hoverTextShadowColor?: string;
    /** Hover image shadow color */
    hoverImgShadowColor?: string;
    /** Pressed Background color */
    pressedBgColor?: string;
    /** Pressed Text color */
    pressedTextColor?: string;
}

export interface StyledProps extends AvatarProps {
    /** To check children is image */
    isImage?: boolean;
}
