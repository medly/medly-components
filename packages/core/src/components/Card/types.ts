import { HTMLProps } from '@medly-components/utils';

export type ItemsPosition = 'top' | 'bottom' | 'left' | 'right' | 'center';
export type Variants = 'solid' | 'flat';
export type FlowDirection = 'vertical' | 'horizontal';

export interface CardProps extends HTMLProps<HTMLDivElement> {
    /** To be used if rendering like Link */
    to?: string;
    /** Card design */
    variant?: 'solid' | 'flat';
    /** Use card as flex container of normal block */
    display?: 'block' | 'flex';
    /** Direction of the children */
    flowDirection?: 'vertical' | 'horizontal';
    /** Separator between each children */
    separator?: boolean;
    /** To be used to define width/height of cards inside cards */
    flex?: number;
    /** Enable padding */
    withoutPadding?: boolean;
    /** Alignment of the card content */
    alignItems?: 'top' | 'bottom' | 'left' | 'right' | 'center';
    /** Takes full width of the parent component */
    fullWidth?: boolean;
    /** Takes full height of the parent component */
    fullHeight?: boolean;
    /** Min width in px/rem/% (1rem = 10px)*/
    minWidth?: string;
    /** Min width in px/rem/% (1rem = 10px) */
    minHeight?: string;
}
