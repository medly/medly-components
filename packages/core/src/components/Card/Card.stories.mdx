import { HTMLProps } from '@medly-components/utils';
import { HomeIcon } from '@medly-components/icons';
import { Card } from './Card.tsx';
import Text from '../Text';
import { withKnobs, text, color, select, boolean } from '@storybook/addon-knobs';
import { Preview, Story, Meta, Props } from '@storybook/addon-docs/blocks';
import { action } from '@storybook/addon-actions';
import { variants, display, itemsPosition, flowDirection, ThemeInterface } from './Card.stories';

<Meta title="Core" component={Card} parameters={{ jest: ['Card.test.tsx'] }} />

# Card


A simple card component with a lot of customization. The hover effect will work only when you pass the `onClick` prop. You can play with the props in the **Canvas** tab. 

<Preview withToolbar>
    <Story name="Card" decorators={[withKnobs]}>
        <Card
            variant={select('Variant', variants, 'solid')}
            flowDirection={select('Flow Direction', flowDirection, 'horizontal')}
            alignItems={select('Items Position', itemsPosition, 'center')}
            display={select('Display', display, 'flex')}
            separator={boolean('Separator', true)}
            withoutPadding={boolean('Without Padding', true)}
            fullWidth={boolean('Full Width', false)}
            fullHeight={boolean('Full Height', false)}
            minWidth={text('Min Width', '')}
            minHeight={text('Min Height', '')}
            onClick={action('Clicked')}
        >
            <Card variant="flat" display="flex" alignItems="center">
                <Text textWeight="Strong" textVariant="h2">
                    8
                </Text>
                <Text>Marks</Text>
            </Card>
            <Card variant="flat" display="flex" flex={6} alignItems="left">
                <Text>17607124</Text>
                <Text textVariant="h3">John, Snow</Text>
                <Text>Son of Lyanna Stark and Rhaegar Targaryen, the late Prince of Dragonstone</Text>
            </Card>
        </Card>
    </Story>
</Preview>

> **Note**: Shadow on hovering will appear only if you pass the `onClick` prop to it.

### Variants

<Preview>
    <Card variant="solid">
        <Text>Solid Card</Text>
    </Card>
    <Card variant="flat">
        <Text>Flat Card</Text>
    </Card>
</Preview>

### Children direction

To customize the card component in many ways, use `display='flex'`.

<Preview>
    <Card display="flex" flowDirection="horizontal" alignItems="center">
        <HomeIcon />
        <Text>Horizontal</Text>
    </Card>
    <Card display="flex" flowDirection="vertical">
        <HomeIcon />
        <Text>Vertical</Text>
    </Card>
</Preview>

### Props

<Props of={Card} />

### CardTheme

<Props of={ThemeInterface} />
