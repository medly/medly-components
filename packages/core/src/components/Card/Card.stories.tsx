import { CardTheme, defaultTheme } from '@medly-components/theme';
import { CardProps } from './types';
import type { FC } from 'react';

export const variants: CardProps['variant'][] = ['solid', 'flat'];
export const display: CardProps['display'][] = ['block', 'flex'];
export const itemsPosition: CardProps['alignItems'][] = ['top', 'bottom', 'left', 'right', 'center'];
export const flowDirection: CardProps['flowDirection'][] = ['vertical', 'horizontal'];

export const ThemeInterface: FC<CardTheme> = () => null;
ThemeInterface.defaultProps = {
    ...defaultTheme.card
};
