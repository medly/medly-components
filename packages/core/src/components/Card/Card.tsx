import { WithStyle } from '@medly-components/utils';
import React, { FC } from 'react';
import * as Styled from './Card.styled';
import { CardProps } from './types';

const Component: FC<CardProps> = React.memo(React.forwardRef((props, ref) => <Styled.Card ref={ref} {...props} />));
Component.defaultProps = {
    display: 'block',
    variant: 'solid',
    flowDirection: 'vertical',
    separator: false,
    flex: 1,
    withoutPadding: false,
    alignItems: 'center',
    fullWidth: false,
    fullHeight: false
};
Component.displayName = 'Card';
export const Card: FC<CardProps> & WithStyle = Object.assign(Component, { Style: Styled.Card });
