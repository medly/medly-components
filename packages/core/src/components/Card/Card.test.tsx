import { render } from '@test-utils';
import React from 'react';
import Text from '../Text';
import { Card } from './Card';
import { CardProps } from './types';

describe('Card component', () => {
    const variants: Required<CardProps>['variant'][] = ['solid', 'flat'],
        displayOptions: Required<CardProps>['display'][] = ['block', 'flex'];
    it('should render correctly with all the default props', () => {
        const { container } = render(
            <Card>
                <Text>Dummy Text</Text>
            </Card>
        );
        expect(container).toMatchSnapshot();
    });

    test.each(variants)('should render properly with %p variant', variant => {
        const { container } = render(
            <Card variant={variant}>
                <Text>Dummy Text</Text>
            </Card>
        );
        expect(container).toMatchSnapshot();
    });

    test.each(displayOptions)('should render properly with display %p', display => {
        const { container } = render(
            <Card display={display}>
                <Text>Dummy Text</Text>
            </Card>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render properly when it is clickable ', () => {
        const { container } = render(
            <Card onClick={jest.fn()}>
                <Text>Dummy Text</Text>
            </Card>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render properly when fullWidth and fullHeight is set true', () => {
        const { container } = render(
            <Card fullWidth fullHeight>
                <Text>Dummy Text</Text>
            </Card>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render properly when flow direction is horizontal and separator is set true', () => {
        const { container } = render(
            <Card separator flowDirection="horizontal">
                <Card separator flowDirection="horizontal" alignItems="left">
                    <Text>Dummy Text</Text>
                </Card>
                <Card separator flowDirection="horizontal" alignItems="right">
                    <Text>Dummy Text</Text>
                </Card>
                <Card separator flowDirection="horizontal" alignItems="top">
                    <Text>Dummy Text</Text>
                </Card>
                <Card separator flowDirection="horizontal" alignItems="bottom">
                    <Text>Dummy Text</Text>
                </Card>
                <Card separator flowDirection="horizontal">
                    <Text>Dummy Text</Text>
                </Card>
            </Card>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render properly when separator is set true flow direction is vertical', () => {
        const { container } = render(
            <Card separator withoutPadding flowDirection="vertical" alignItems="left">
                <Card variant="flat" flowDirection="vertical" alignItems="left">
                    <Text>Marks</Text>
                </Card>
                <Card variant="flat" flex={6} flowDirection="vertical" alignItems="right">
                    <Text>Dummy Text</Text>
                </Card>
                <Card variant="flat" flex={6} flowDirection="vertical" alignItems="top">
                    <Text>Dummy Text</Text>
                </Card>
                <Card variant="flat" flex={6} flowDirection="vertical" alignItems="bottom">
                    <Text>Dummy Text</Text>
                </Card>
            </Card>
        );
        expect(container).toMatchSnapshot();
    });
});
