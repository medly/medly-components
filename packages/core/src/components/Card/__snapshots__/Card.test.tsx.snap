// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render correctly with all the default props 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c0 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: inline-block;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  background-color: #ffffff;
  border-radius: 0.8rem;
  box-shadow: 0 0.2rem 0.8rem rgba(176,188,200,0.2);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:0px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:768px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:1024px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:1280px) {
  .c0 {
    padding: 2.4rem;
  }
}

<div>
  <div
    class="c0"
    display="block"
  >
    <span
      class="c1"
    >
      Dummy Text
    </span>
  </div>
</div>
`;

exports[`Card component should render properly when flow direction is horizontal and separator is set true 1`] = `
.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c0 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: inline-block;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  background-color: #ffffff;
  border-radius: 0.8rem;
  box-shadow: 0 0.2rem 0.8rem rgba(176,188,200,0.2);
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > div + div {
  border-left: 1px solid #98A7B7;
}

.c1 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: inline-block;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  background-color: #ffffff;
  border-radius: 0.8rem;
  box-shadow: 0 0.2rem 0.8rem rgba(176,188,200,0.2);
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 > div + div {
  border-left: 1px solid #98A7B7;
}

.c3 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: inline-block;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  background-color: #ffffff;
  border-radius: 0.8rem;
  box-shadow: 0 0.2rem 0.8rem rgba(176,188,200,0.2);
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c3 > div + div {
  border-left: 1px solid #98A7B7;
}

.c4 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: inline-block;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  background-color: #ffffff;
  border-radius: 0.8rem;
  box-shadow: 0 0.2rem 0.8rem rgba(176,188,200,0.2);
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 > div + div {
  border-left: 1px solid #98A7B7;
}

.c5 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: inline-block;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  background-color: #ffffff;
  border-radius: 0.8rem;
  box-shadow: 0 0.2rem 0.8rem rgba(176,188,200,0.2);
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 > div + div {
  border-left: 1px solid #98A7B7;
}

@media (min-width:0px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:768px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:1024px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:1280px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:0px) {
  .c1 {
    padding: 2.4rem;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 2.4rem;
  }
}

@media (min-width:1024px) {
  .c1 {
    padding: 2.4rem;
  }
}

@media (min-width:1280px) {
  .c1 {
    padding: 2.4rem;
  }
}

@media (min-width:0px) {
  .c3 {
    padding: 2.4rem;
  }
}

@media (min-width:768px) {
  .c3 {
    padding: 2.4rem;
  }
}

@media (min-width:1024px) {
  .c3 {
    padding: 2.4rem;
  }
}

@media (min-width:1280px) {
  .c3 {
    padding: 2.4rem;
  }
}

@media (min-width:0px) {
  .c4 {
    padding: 2.4rem;
  }
}

@media (min-width:768px) {
  .c4 {
    padding: 2.4rem;
  }
}

@media (min-width:1024px) {
  .c4 {
    padding: 2.4rem;
  }
}

@media (min-width:1280px) {
  .c4 {
    padding: 2.4rem;
  }
}

@media (min-width:0px) {
  .c5 {
    padding: 2.4rem;
  }
}

@media (min-width:768px) {
  .c5 {
    padding: 2.4rem;
  }
}

@media (min-width:1024px) {
  .c5 {
    padding: 2.4rem;
  }
}

@media (min-width:1280px) {
  .c5 {
    padding: 2.4rem;
  }
}

<div>
  <div
    class="c0"
    display="block"
  >
    <div
      class="c1"
      display="block"
    >
      <span
        class="c2"
      >
        Dummy Text
      </span>
    </div>
    <div
      class="c3"
      display="block"
    >
      <span
        class="c2"
      >
        Dummy Text
      </span>
    </div>
    <div
      class="c4"
      display="block"
    >
      <span
        class="c2"
      >
        Dummy Text
      </span>
    </div>
    <div
      class="c5"
      display="block"
    >
      <span
        class="c2"
      >
        Dummy Text
      </span>
    </div>
    <div
      class="c0"
      display="block"
    >
      <span
        class="c2"
      >
        Dummy Text
      </span>
    </div>
  </div>
</div>
`;

exports[`Card component should render properly when fullWidth and fullHeight is set true 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c0 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: block;
  width: 100%;
  height: 100%;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  background-color: #ffffff;
  border-radius: 0.8rem;
  box-shadow: 0 0.2rem 0.8rem rgba(176,188,200,0.2);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:0px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:768px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:1024px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:1280px) {
  .c0 {
    padding: 2.4rem;
  }
}

<div>
  <div
    class="c0"
    display="block"
  >
    <span
      class="c1"
    >
      Dummy Text
    </span>
  </div>
</div>
`;

exports[`Card component should render properly when it is clickable 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c0 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: inline-block;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  background-color: #ffffff;
  border-radius: 0.8rem;
  box-shadow: 0 0.2rem 0.8rem rgba(176,188,200,0.2);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0,
.c0 * {
  cursor: pointer;
}

.c0:hover {
  box-shadow: 0 0.4rem 0.8rem rgba(176,188,200,0.6);
}

@media (min-width:0px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:768px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:1024px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:1280px) {
  .c0 {
    padding: 2.4rem;
  }
}

<div>
  <div
    class="c0"
    display="block"
  >
    <span
      class="c1"
    >
      Dummy Text
    </span>
  </div>
</div>
`;

exports[`Card component should render properly when separator is set true flow direction is vertical 1`] = `
.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c0 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: inline-block;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  background-color: #ffffff;
  border-radius: 0.8rem;
  box-shadow: 0 0.2rem 0.8rem rgba(176,188,200,0.2);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c0 > div + div {
  border-top: 1px solid #98A7B7;
}

.c1 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: inline-block;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  border-radius: 0;
  margin: 0;
  border: none;
  box-shadow: none;
  background-color: transparent;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c3 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 6;
  -ms-flex: 6;
  flex: 6;
  display: inline-block;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  border-radius: 0;
  margin: 0;
  border: none;
  box-shadow: none;
  background-color: transparent;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c4 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 6;
  -ms-flex: 6;
  flex: 6;
  display: inline-block;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  border-radius: 0;
  margin: 0;
  border: none;
  box-shadow: none;
  background-color: transparent;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 6;
  -ms-flex: 6;
  flex: 6;
  display: inline-block;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  border-radius: 0;
  margin: 0;
  border: none;
  box-shadow: none;
  background-color: transparent;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:0px) {
  .c1 {
    padding: 2.4rem;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 2.4rem;
  }
}

@media (min-width:1024px) {
  .c1 {
    padding: 2.4rem;
  }
}

@media (min-width:1280px) {
  .c1 {
    padding: 2.4rem;
  }
}

@media (min-width:0px) {
  .c3 {
    padding: 2.4rem;
  }
}

@media (min-width:768px) {
  .c3 {
    padding: 2.4rem;
  }
}

@media (min-width:1024px) {
  .c3 {
    padding: 2.4rem;
  }
}

@media (min-width:1280px) {
  .c3 {
    padding: 2.4rem;
  }
}

@media (min-width:0px) {
  .c4 {
    padding: 2.4rem;
  }
}

@media (min-width:768px) {
  .c4 {
    padding: 2.4rem;
  }
}

@media (min-width:1024px) {
  .c4 {
    padding: 2.4rem;
  }
}

@media (min-width:1280px) {
  .c4 {
    padding: 2.4rem;
  }
}

@media (min-width:0px) {
  .c5 {
    padding: 2.4rem;
  }
}

@media (min-width:768px) {
  .c5 {
    padding: 2.4rem;
  }
}

@media (min-width:1024px) {
  .c5 {
    padding: 2.4rem;
  }
}

@media (min-width:1280px) {
  .c5 {
    padding: 2.4rem;
  }
}

<div>
  <div
    class="c0"
    display="block"
  >
    <div
      class="c1"
      display="block"
    >
      <span
        class="c2"
      >
        Marks
      </span>
    </div>
    <div
      class="c3"
      display="block"
    >
      <span
        class="c2"
      >
        Dummy Text
      </span>
    </div>
    <div
      class="c4"
      display="block"
    >
      <span
        class="c2"
      >
        Dummy Text
      </span>
    </div>
    <div
      class="c5"
      display="block"
    >
      <span
        class="c2"
      >
        Dummy Text
      </span>
    </div>
  </div>
</div>
`;

exports[`Card component should render properly with "flat" variant 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c0 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: inline-block;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  border-radius: 0;
  margin: 0;
  border: none;
  box-shadow: none;
  background-color: transparent;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:0px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:768px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:1024px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:1280px) {
  .c0 {
    padding: 2.4rem;
  }
}

<div>
  <div
    class="c0"
    display="block"
  >
    <span
      class="c1"
    >
      Dummy Text
    </span>
  </div>
</div>
`;

exports[`Card component should render properly with "solid" variant 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c0 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: inline-block;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  background-color: #ffffff;
  border-radius: 0.8rem;
  box-shadow: 0 0.2rem 0.8rem rgba(176,188,200,0.2);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:0px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:768px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:1024px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:1280px) {
  .c0 {
    padding: 2.4rem;
  }
}

<div>
  <div
    class="c0"
    display="block"
  >
    <span
      class="c1"
    >
      Dummy Text
    </span>
  </div>
</div>
`;

exports[`Card component should render properly with display "block" 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c0 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: inline-block;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  background-color: #ffffff;
  border-radius: 0.8rem;
  box-shadow: 0 0.2rem 0.8rem rgba(176,188,200,0.2);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:0px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:768px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:1024px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:1280px) {
  .c0 {
    padding: 2.4rem;
  }
}

<div>
  <div
    class="c0"
    display="block"
  >
    <span
      class="c1"
    >
      Dummy Text
    </span>
  </div>
</div>
`;

exports[`Card component should render properly with display "flex" 1`] = `
.c1 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c0 {
  overflow: auto;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  background-color: #ffffff;
  border-radius: 0.8rem;
  box-shadow: 0 0.2rem 0.8rem rgba(176,188,200,0.2);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:0px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:768px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:1024px) {
  .c0 {
    padding: 2.4rem;
  }
}

@media (min-width:1280px) {
  .c0 {
    padding: 2.4rem;
  }
}

<div>
  <div
    class="c0"
    display="flex"
  >
    <span
      class="c1"
    >
      Dummy Text
    </span>
  </div>
</div>
`;
